### ISSUE TITLE
Shows "Generating keys. This may take a while." and gets stuck there.

### ISSUE THREAD

**Issue by digate** [syncthing/syncthing-android#216]
Hi! I'm new here. I've installed Syncthing in my Samsung Galaxy S2 (rooted) today and it shows the message "Generating keys. This may take a while." and doesn't go forward from there, just shows the spinning circle. When I restart the phone and open Syncthing it shows "Waiting for API" and gets stuck again. Should I paste the log here or something ? (In that case, how do I get the log?) Thanks!

**Comment by calmh** [comment]
That may be a long operation. On a slow CPU, minutes probably.

**Comment by Zillode** [comment]
Yes, please post your logs. Read [this](https://github.com/syncthing/syncthing-android/blob/master/CONTRIBUTING.md#bug-reports) to obtain it.

**Comment by digate** [comment]
@calmh Even more than 5 hours? Is it possible?
Here are my logs:

syncthing-android version: 0.5.17
Samsung Galaxy S2 GT-I9100 Android v4.1.2

logcat: http://pastebin.com/nGPK4Rt3
config.xml: http://pastebin.com/VmjNbvTu

**Comment by AudriusButkevicius** [comment]
the fact that you have a config implies that you managed to get the keys generated.

**Comment by Nutomic** [comment]
@digate There's nothing useful in the log. Can you reboot the device, then start the app and get a log?

**Comment by chucic** [comment]
Hi, i run into this exactly same while ago, today found this thread and decided i would try to obtain log.

After some time (provided howto isn't much usefull anymore, since you have to be root to use an app and from terminal it didn't work either for me) I almost give up, I don't need syncthing on android that much. But you know, unresolved issue, so I got back to it. At the end I had to do it from PC with Android SDK.

But at the and, i got my 252313 lines of log, then i searched it and found that keys were generated, GUI started and everything seemed to be working, but syncthing still showed "generating...". Only some complains from proxy > https > adblock > false 127.0.0.1:8080, so i unset proxy (localhost:2020 used by adblock) from my current wifi connection and voila, syncthing app is working.

Even if i turn on proxy again, sycnting is still working, cpu usage changing... And when i shutdown syncthing and start it without disabling proxy (so i got endless "Waiting for API") web gui works normally. Since to adblock in android is quite hard to ad custom filters (and requires root anyway), i have no chance test if i could somehow bypass that.

And to web gui i had to connect to 127.0.0.1:37534, but log still says 127.0.0.1:8080...

**Comment by Nutomic** [comment]
Ok I'm getting this myself sometimes, haven't found why.

A device restart should work around it.

**Comment by syadykin** [comment]
Well, phone restart did not help in my case. I found in log that app asks for execution permission for unknown reason and after chmod it to 755 it run ok:

     mount -o rw,remount /dev/block/dm-38 /mnt/asec/com.nutomic.syncthingandroid-1
     chmod 755 /mnt/asec/com.nutomic.syncthingandroid-1/lib/libsyncthing.so

and restart app then. Phone restart causes this issue again (waiting for API)

Phone: Lenovo P780
Android version: 4.4.2
Firmware: P780_ROW_8G_S228_150112

**Comment by Nutomic** [comment]
That's weird, that file should always be executable. Also, it should normally be in `com.nutomic.syncthingandroid/lib/`.

**Comment by Nutomic** [comment]
It seems I'm also getting this problem on a Nexus S. The log only shows `Generating RSA key and certificate for syncthing...`, and the data folder remains empty.

I also didn't change anything around the app start, so I assume this is a problem with syncthing.

**Comment by Zillode** [comment]
I'm pretty sure this is due to the [async call](https://github.com/syncthing/syncthing-android/blob/master/src/main/java/com/nutomic/syncthingandroid/syncthing/RestApi.java#L270) shuttting down the Syncthing service when it should be running. [This PR](
https://github.com/syncthing/syncthing-android/pull/321) fixes it.

**Comment by kingdave** [comment]
I see same problem as *syadykin* - bad permissions on that file. Seems solved by changing the permissions as he said.

**Comment by Zillode** [comment]
Relevant: info by @marcelpaulo https://bpaste.net/show/694ae544fd28

**Comment by Zillode** [comment]
(to check: what happens if wifi is down when first launching the app)
